### PR TITLE
add filter test fixture with supporting unit tests

### PR DESCRIPTION
**Pull request by johnhwhite**
This PR addresses issue #186 

* Add test fixture for filter component

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-lists?_sp.spa=pr-186-filter-test-fixture](https://developer.blackbaud.com/skyux-lists?_sp.spa=pr-186-filter-test-fixture)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-lists?_sp.spa=pr-186-filter-test-fixture](https://developer.blackbaud.com/skyux-lists?_sp.spa=pr-186-filter-test-fixture)

**Review by Blackbaud-AlexKingman — COMMENTED** [review]
Looking good. Left a few suggestions, and want to pose a thought -

Because we have several inputs on the `sky-filter-button` component (`ariaLabel`, `ariaControls`, `disabled`, etc...), how would a consumer easily get to those elements? I'm wondering if we should create a type for `SkyListsFixtureFilterButton` and have primitive properties on it that return these pieces. [Take a look at the pattern we created for dropdown fixture](https://github.com/blackbaud/skyux-popovers/blob/master/src/app/public/testing/dropdown/dropdown-fixture.ts#L126).

Notice how `SkyPopoversFixtureDropdownItem` and `SkyPopoversFixtureDropdownMenu` are just simple interfaces that get populated by querying the HTML attributes in the DOM. In addition to returning the attributes that correlate with the component's inputs, [you could also return the `buttonText` and `filterItemText` so the consumer can ensure their text is getting populated in the buttons](https://github.com/blackbaud/skyux-popovers/blob/master/src/app/public/testing/dropdown/dropdown-fixture.ts#L58).

Should follow this patter for both the button and filter items?

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-lists?_sp.spa=pr-186-filter-test-fixture](https://developer.blackbaud.com/skyux-lists?_sp.spa=pr-186-filter-test-fixture)

File: src/app/public/testing/filter/filter-fixture.spec.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+      expect(testComponent.buttonClicked).toBeTrue();
+    });
+
+    it('should do nothing when the button is not enabled', () => {
+      testComponent.buttonIsDisabled = true;
+      expect(testComponent.buttonClicked).toBeFalse();
+      fixture.detectChanges();
+      return filterButtonFixture.clickFilterButton().then(() => {
+        fixture.detectChanges();
+        expect(testComponent.buttonClicked).toBeFalse();
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/testing/filter/filter-fixture.spec.ts:151]: Just for future reference, another way you could check for a click without having to store a value in your component is to just spy on the event method.
```
const spy = spyOn(testComponent, 'filterButtonClicked');
expect(spy).toHaveBeenCalled();
```
Just a suggestion, but not holding up this PR.

File: src/app/public/testing/filter/filter-fixture-summary.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import {
+  ComponentFixture
+} from '@angular/core/testing';
+
+import {
+  SkyAppTestUtility
+} from '@skyux-sdk/testing';
+
+/**
+ * Provides information for and interaction with a SKY UX filter button component.
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/testing/filter/filter-fixture-summary.ts:14]: Provides information for and interaction with a SKY UX `filter summary` component.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-lists?_sp.spa=pr-186-filter-test-fixture](https://developer.blackbaud.com/skyux-lists?_sp.spa=pr-186-filter-test-fixture)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-lists?_sp.spa=pr-186-filter-test-fixture](https://developer.blackbaud.com/skyux-lists?_sp.spa=pr-186-filter-test-fixture)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-lists?_sp.spa=pr-186-filter-test-fixture](https://developer.blackbaud.com/skyux-lists?_sp.spa=pr-186-filter-test-fixture)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-lists?_sp.spa=pr-186-filter-test-fixture](https://developer.blackbaud.com/skyux-lists?_sp.spa=pr-186-filter-test-fixture)

File: src/app/public/testing/filter/filter-fixture.spec.ts [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+      fixture.detectChanges();
+      await filterButtonFixture.clickFilterButton();
+      fixture.detectChanges();
+      expect(testComponent.buttonClicked).toBeFalse();
+    });
+
+    it('should get the button aria controls', async () => {
+      fixture.detectChanges();
+      await fixture.whenStable();
+      expect(filterButtonFixture.button.ariaControls).toBeFalsy();
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/testing/filter/filter-fixture.spec.ts:155]: Are you concerned that checking for falsy might miss something? By default, `ariaControls` doesn't have a value and is therefore falsy. If this breaks, we might not know. Instead, could we set `[ariaControls]="'somevalue'"` in your fixture template, and then test for that value here? `expect(filterButtonFixture.button.ariaControls).toEqual('somevalue');`
I'd also check the same for `ariaExpanded` below ...

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-lists?_sp.spa=pr-186-filter-test-fixture](https://developer.blackbaud.com/skyux-lists?_sp.spa=pr-186-filter-test-fixture)

**Review by Blackbaud-AlexKingman — APPROVED** [review]

**Pull request merged**